app/src/main/java/io/gnosis/safe/ui/transactions/details/view/LabeledArrayItem.kt

```diff
@@ -65,38 +65,38 @@ class LabeledArrayItem @JvmOverloads constructor(
                         addValueItem(arrayItem.container, context.getString(R.string.array), ParamType.VALUE)
                     }
                 } else {
-                    addValueItem(arrayItem.container, it as String, paramType)
+                    addValueItem(arrayItem.container, it, paramType)
                 }
             }
         }
         nestingLevel--
         container.addView(arrayItem)
     }
 
-    private fun addValueItem(container: ViewGroup, value: String, paramType: ParamType) {
+    private fun addValueItem(container: ViewGroup, value: Any, paramType: ParamType) {
         when (paramType) {
             ParamType.ADDRESS -> {
                 val addressItem = AddressItem(context)
                 val layoutParams = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
                 layoutParams.setMargins(0, dpToPx(8), 0, 0)
                 addressItem.layoutParams = layoutParams
-                addressItem.address = value.asEthereumAddress()
+                addressItem.address = value as Solidity.Address
                 container.addView(addressItem)
             }
             ParamType.BYTES -> {
                 val bytesItem = TxDataView(context)
                 val layoutParams = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
                 layoutParams.setMargins(0, dpToPx(6), 0, 0)
                 bytesItem.layoutParams = layoutParams
-                bytesItem.setData(value, value.length / 2)
+                bytesItem.setData(value as String, value.length / 2)
```

Comment:
The length calculation might be off, if there is a `0x` prefix. Assuming value is already a hex string.